crates/vm/src/system/memory/smallvec.rs

```diff
@@ -0,0 +1,81 @@
+use std::{
+    fmt::Debug,
+    ops::{Deref, DerefMut},
+};
+
+use serde::{Deserialize, Serialize};
+use smallvec::{Array, SmallVec};
+
+pub struct SerdeSmallVec<A: Array>(SmallVec<A>);
```

Comment:
is this purely for serde? can https://crates.io/crates/serde_with help with this

Reply:
Yes. I had tried using `serde_with`, but ran into issues because `SmallVec` cannot be directly converted into `Vec`.